control-plane/connect-inject/endpoints_controller_test.go

```diff
@@ -886,6 +886,7 @@ func TestReconcileCreateEndpoint(t *testing.T) {
 				pod1.Annotations[annotationService] = "different-consul-svc-name"
 				pod1.Annotations[fmt.Sprintf("%sname", annotationMeta)] = "abc"
 				pod1.Annotations[fmt.Sprintf("%sversion", annotationMeta)] = "2"
+				pod1.Annotations[fmt.Sprintf("%spod_name", annotationMeta)] = "$POD_NAME"
 				pod1.Annotations[annotationTags] = "abc,123,$POD_NAME"
```

Comment:
I see two tests referencing `$POD_NAME` did this never work to begin with or are the tests bogus?

Reply:
I think there was a `pod-name` vs `pod_name` situation with the keys. Additionally, i think the provide the env var `$POD_NAME` use different annotations and we are supporting both.

Reply:
ahhh, that makes sense!